src/mozillavpn.cpp

```diff
@@ -859,8 +859,18 @@ void MozillaVPN::logout() {
   }
 
   if (m_private->m_deviceModel.hasCurrentDevice(keys())) {
-    TaskScheduler::scheduleTask(new TaskRemoveDevice(keys()->publicKey()));
+    TaskScheduler::scheduleTask(new TaskGroup(
+        {new TaskRemoveDevice(keys()->publicKey()),
+         // Immediately after the scheduling of the device removal, we want to
+         // delete the session token, so that, in case the app is terminated, at
+         // the next execution we go back to the init screen.
+         new TaskFunction([this]() { reset(false); })}));
+
+    // In case the app is closed even before scheduling the previous TaskGroup,
+    // removing the key we will enforce a new authentication at the first
+    // TaskAccount execution.
```

Comment:
Why is this logic only necessary when the user has the current device? What if the app is closed even before scheduling the reset task on line 876?

Reply:
Nevermind, I get it on a second look.